packages/grid/x-data-grid/src/hooks/features/filter/gridFilterState.ts

```diff
@@ -59,3 +53,12 @@ export interface GridFilteringMethodParams {
 }
 
 export type GridFilteringMethodValue = Omit<GridFilterState, 'filterModel'>;
+
+/**
+ * Visibility status for each row.
+ * A row is visible if it is passing the filters AND if its parents are expanded.
+ * If a row is not registered in this lookup, it is visible.
+ */
+export interface GridVisibleRowsLookupState extends Record<GridRowId, boolean> {}
```

Comment:
Just fyi, you may have seen in my filtering PR, but I was hoping to convert both the filteredRows & visibleRows from objects into `Set`s as it improves performance for large datasets.

Minor note on the type, I would find `type GridVisibleRowsLookupState = Record<GridRowId, boolean>` more semantically accurate. We're not building a new interface, we're defining what the shape of that state is, and it _is_ a record.

Reply:
> Just fyi, you may have seen in my filtering PR, but I was hoping to convert both the filteredRows & visibleRows from objects into Sets as it improves performance for large datasets.

Yes, but that would be a breaking change, right?
Unless we do something like this:
```tsx
/**
 * For backward compatibility only, in case if anyone is using `gridVisibleRowsLookupSetSelector`
 * @deprecated Use `gridVisibleRowsLookupSetSelector` instead
 */
export const gridVisibleRowsLookupSelector = (state: GridStateCommunity) => setToObject(state.visibleRowsLookup);

// TODO v7: rename to `gridVisibleRowsLookupSelector`
export const gridVisibleRowsLookupSetSelector = (state: GridStateCommunity) => state.visibleRowsLookup;


```

Reply:
Yes for the change, that's what I'm planning to do for the selectors.